e2e/env/env.go

```diff
@@ -86,50 +86,66 @@ func setupClient(kubeconfigPath string) (client.Client, error) {
 }
 
 func New(config *types.Config, log *zap.SugaredLogger) (*types.Env, error) {
-	var err error
-
 	env := &types.Env{}
 
-	env.Hub.Name = config.Clusters["hub"].Name
-	if env.Hub.Name == "" {
-		env.Hub.Name = defaultHubClusterName
-		log.Infof("Cluster \"hub\" name not set, using default name %q", defaultHubClusterName)
+	if err := setupHub(&env.Hub, config.Clusters["hub"], defaultHubClusterName, log); err != nil {
+		return nil, fmt.Errorf("failed to setup hub cluster \"hub\": %w", err)
 	}
 
-	env.Hub.Client, err = setupClient(config.Clusters["hub"].Kubeconfig)
-	if err != nil {
-		return nil, fmt.Errorf("failed to create clients for hub cluster: %w", err)
+	if err := setupManagedCluster(&env.C1, config.Clusters["c1"], defaultC1ClusterName, log); err != nil {
+		return nil, fmt.Errorf("failed to setup managed cluster \"c1\": %w", err)
 	}
 
-	env.Hub.Kubeconfig = config.Clusters["hub"].Kubeconfig
-
-	env.C1.Name = config.Clusters["c1"].Name
-	if env.C1.Name == "" {
-		env.C1.Name = defaultC1ClusterName
-		log.Infof("Cluster \"c1\" name not set, using default name %q", defaultC1ClusterName)
+	if err := setupManagedCluster(&env.C2, config.Clusters["c2"], defaultC2ClusterName, log); err != nil {
+		return nil, fmt.Errorf("failed to setup managed cluster \"c2\": %w", err)
 	}
 
-	env.C1.Client, err = setupClient(config.Clusters["c1"].Kubeconfig)
+	return env, nil
+}
+
+func setupHub(
+	cluster *types.Cluster, clusterConfig types.ClusterConfig,
+	defaultName string, log *zap.SugaredLogger,
```

Comment:
If we cannot fit the arguments in one line, use one line per argument. Your editor probably provide this formatting option.